fastscanner/adapters/realtime/redis_channel.py

```diff
@@ -0,0 +1,46 @@
+import logging
+from typing import Any, cast
+
+import pandas as pd
+import redis.asyncio as aioredis
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class RedisChannel:
+    def __init__(
+        self,
+        host: str,
+        port: int,
+        password: str | None,
+        db: int,
+    ):
+        # self.redis = aioredis.Redis(
+        #     host=host,
+        #     port=port,
+        #     password=password,
+        #     db=db,
+        #     decode_responses=True,
+        # )
+        self.redis = aioredis.Redis(
+            unix_socket_path="/run/redis/redis-server.sock",
+            password=None,
+            db=0,
+            decode_responses=True,
+        )
+        self._pipeline = None
+
```

Comment:
We can now implement a 
```
async def subscribe(self, channel_id: str, handler: "ChannelHandler"): ...
```

Then you can test in the benchmark_redis_read.py with a simple ChannelHandler
```
 class BenchmarkChannelHandler(Protocol):
    async def handle(self, channel_id: str, data: dict[Any, Any]):
            # here all the code you have in the benchmark_redis_read.py after the xread.
```